composer/datasets/ade20k.py

```diff
@@ -388,6 +391,155 @@ def initialize_object(self, batch_size, dataloader_hparams) -> DataSpec:
                         device_transforms=device_transform_fn)
 
 
+class StreamingADE20k(StreamingDataset):
+    """
+    Implementation of the ADE20k dataset using StreamingDataset.
+
+    Args:
+        remote (str): Remote directory (S3 or local filesystem) where dataset is stored.
+        local (str): Local filesystem directory where dataset is cached during operation.
+        split (str): The dataset split to use, either 'train', or 'val'. Default: ``'train```.
+        shuffle (bool): Whether to shuffle the samples in this dataset.
+        base_size (int): initial size of the image and target before other augmentations. Default: ``512``.
+        min_resize_scale (float): the minimum value the samples can be rescaled. Default: ``0.5``.
+        max_resize_scale (float): the maximum value the samples can be rescaled. Default: ``2.0``.
+        final_size (int): the final size of the image and target. Default: ``512``.
+    """
```

Comment:
Refactored so that the `StreamingADE20k` dataset does most of the work, making it easier to use in a standalone way.

Collating and device normalization are still done in the hparams `initialize_object` when we create the dataloader.

@hanlint Can I avoid refactoring the older `ImageFolder`-based ADE20k dataset in this PR? I an open a JIRA issue to track. Would prefer to make this PR simple and additive, and do edits/deletions in a separate one.

Reply:
Yeah OK to limit your refactor to streaming datasets